examples/audio_server/server.cc

```diff
@@ -0,0 +1,132 @@
+    // Need to join and remove finished sessions
+    auto joiner = std::thread([&threads, &safe_state]{
+      while(true) {
+        std::vector<std::string> to_join;
+        for (auto& pair : threads) {
+          std::string channel_id = pair.first;
+          auto has_channel = safe_state.MutableUse([channel_id](SharedState& state) {
+            return state.channel_exists(channel_id);
+          });
+          if (!has_channel) {
+            to_join.push_back(pair.first);
+            pair.second.join();
+          }
+        }
+        for(auto& to_drop: to_join){
+          threads.erase(to_drop);
+        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+      }
+    });
+
+    auto scope = http.Register("/", [&safe_state, &threads](Request r) {
+      std::string channel_id = r.body;
+
+      auto has_channel = safe_state.MutableUse([channel_id](SharedState& state) { return state.channel_exists(channel_id); });
+      if(has_channel){
+        r("error: channel already exists\n");
+        return;
+      }
+      uint32_t n_chans = safe_state.MutableUse([](SharedState& state) { return state.active_channels; });
+      if(n_chans == FLAGS_n){
+        r("error: too many channels\n");
+        return;
+      }
+
+      // Create new channel
+      safe_state.MutableUse([channel_id](SharedState& state) {
+        state.channel_control[channel_id] = "";
+        state.active_channels++;
+      });
+      threads[channel_id] = std::thread([&safe_state, channel_id]() {
+        std::cout << "Channel '" << channel_id << "' is online" << std::endl;
+        while(true) {
+          auto control = safe_state.WaitFor(
+              [channel_id](SharedState const& state) {
+                return state.channel_exists(channel_id);
+              },
+              [channel_id](SharedState & state) {
+                if (state.channel_control[channel_id] == cmd_kill){
+                  state.channel_control.erase(channel_id);
+                  state.active_channels--;
+                  return ControlSignal{"", true};
+                }
+                return ControlSignal{state.channel_control[channel_id], false};
+              },
+              // TODO: reduce this time (only for debug)
+              std::chrono::seconds(1));
+          if(control.stop) {
+            std::cout << "Channel '" << channel_id << "' has been stopped" << std::endl;
+            break;
+          }
+
+          if(!control.apply.empty()) {
+            std::cout << channel_id << "received signal " << control.apply << std::endl;
+            // TODO: handle control signals here
+          }
+
+          // TODO: route the stream here
+          std::cout << "[" << channel_id << "] worker tick" << std::endl;
+
+          std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+        }
+      });
+      r("created\n");
+    });
+
+    scope += http.Register("/kill", [&safe_state](Request r) {
+      std::string channel_id = r.body;
```

Comment:
Maybe `URLPathArgs::CountMask::One` and use `r.url_path_args[0]`, for `localhost:8000/kill/{channel_id}`?

Reply:
Although I've contradicted this comment above, and think `if (r.method == "DELETE")` would be cleaner!

Reply:
Cool idea - just to have /<channel> url with post/delete methods